2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_queue/LinkedQueueTest.java

```diff
@@ -1,103 +1,382 @@
-        assertEquals(1312, integerQueue.poll().intValue());
+        assertThat(hasHeadField).isTrue();
+        assertThat(hasSizeFiled).isTrue();
+        assertThat(hasTailFiled).isTrue();
     }
 
     @Test
     @Order(4)
-    void addElement() {
-        integerQueue.add(324);
-        integerQueue.add(23);
-        integerQueue.add(5);
+    void checkSizeInEmptyQueue() {
+        int size = getInternalSize();
 
-        assertEquals(3, integerQueue.size());
-        assertEquals(324, integerQueue.poll().intValue());
+        assertThat(size).isEqualTo(0);
     }
 
     @Test
     @Order(5)
-    void pollElementFromEmptyQueue() {
-        assertNull(integerQueue.poll());
+    void riseSizeAfterAdd() {
+        addIntElementToQueue(1);
+        addIntElementToQueue(2);
+        addIntElementToQueue(3);
+        int size = getInternalSize();
+
+        assertThat(size).isEqualTo(3);
     }
 
     @Test
-    @Order(6)
-    void pollElement() {
-        integerQueue.add(33);
-        integerQueue.add(123);
-        integerQueue.add(222);
-        integerQueue.add(444);
+    @Order(5)
+    void expenseSizeAfterPoll() {
+        addIntElementToQueue(1);
+        addIntElementToQueue(2);
+        addIntElementToQueue(3);
+        addIntElementToQueue(4);
+
+        int sizeBeforePollOperation = getInternalSize();
+        pollElementFromQueue();
+        pollElementFromQueue();
+        int sizeAfterPollOperation = getInternalSize();
+
+        assertThat(sizeBeforePollOperation).isEqualTo(4);
+        assertThat(sizeAfterPollOperation).isEqualTo(2);
+    }
 
-        integerQueue.poll(); // should poll 33
+    @Test
+    @Order(6)
+    void isEmptyWhenQueueIsEmpty() {
+        boolean isEmpty = this.integerQueue.isEmpty();
+        boolean isEmptyByReflection = isEmptyQueue();
 
-        assertEquals(123, integerQueue.poll().intValue());
-        assertEquals(2, integerQueue.size());
+        assertThat(isEmpty).isEqualTo(isEmptyByReflection);
     }
 
     @Test
     @Order(7)
-    void pollLastElement() {
-        integerQueue.add(8);
-        integerQueue.add(123);
-        integerQueue.add(99);
-        integerQueue.add(46);
+    void isEmptyAfterPollLastElement() {
+        addIntElementToQueue(1);
+        addIntElementToQueue(2);
 
-        integerQueue.poll(); // should poll 8
-        integerQueue.poll(); // should poll 123
-        integerQueue.poll(); // should poll 99
+        boolean isEmptyBeforePoll = isEmptyQueue();
+        pollElementFromQueue();
+        pollElementFromQueue();
+        boolean isEmptyAfterPoll = isEmptyQueue();
 
-        assertEquals(46, integerQueue.poll().intValue());
-        assertEquals(0, integerQueue.size());
+        assertThat(isEmptyBeforePoll).isEqualTo(false);
+        assertThat(isEmptyAfterPoll).isEqualTo(true);
     }
 
+
     @Test
     @Order(8)
-    void isEmptyOnEmptyQueue() {
-        assertTrue(integerQueue.isEmpty());
+    void pollFromEmptyQueue() {
```

Comment:
Lets adhere to tests name convention like this: `methodNameDoesSomethingWhenSomeCondition`.